src/sinks/hdfs/config.rs

```diff
@@ -0,0 +1,190 @@
+use codecs::{encoding::Framer, JsonSerializerConfig, NewlineDelimitedEncoderConfig};
+use opendal::{layers::LoggingLayer, services::Hdfs, Operator};
+use vector_config::configurable_component;
+use vector_core::{
+    config::{AcknowledgementsConfig, DataType, Input},
+    sink::VectorSink,
+};
+
+use crate::{
+    codecs::{Encoder, EncodingConfigWithFraming, SinkType},
+    config::{GenerateConfig, SinkConfig, SinkContext},
+    sinks::{
+        opendal_common::*,
+        util::{
+            partitioner::KeyPartitioner, BatchConfig, BulkSizeBasedDefaultBatchSettings,
+            Compression,
+        },
+        Healthcheck,
+    },
+};
+
+/// Configuration for the `hdfs` sink.
```

Comment:
**praise**: This comment is _awesome_ will definitely point to this as the level we'd like to get everything to! ❤️